java/cuvs-java/src/test/java/com/nvidia/cuvs/BruteForceRandomizedIT.java

```diff
@@ -98,31 +97,44 @@ private void tmpResultsTopKWithRandomValues() throws Throwable {
       }
     }
     // Sanity checks
-    assert dataset.length > 0 : "Dataset is empty.";
+    assert vectors.length > 0 : "Dataset is empty.";
     assert queries.length > 0 : "Queries are empty.";
     assert dimensions > 0 : "Invalid dimensions.";
     assert topK > 0 && topK <= datasetSize : "Invalid topK value.";
 
     // Generate expected results using brute force
-    List<List<Integer>> expected = generateExpectedResults(topK, dataset, queries, prefilters, log);
+    List<List<Integer>> expected = generateExpectedResults(topK, vectors, queries, prefilters, log);
 
     // Create CuVS index and query
     try (CuVSResources resources = CuVSResources.create()) {
 
       BruteForceQuery query = new BruteForceQuery.Builder()
           .withTopK(topK)
           .withQueryVectors(queries)
-          .withPrefilter(prefilters, dataset.length)
+          .withPrefilter(prefilters, vectors.length)
           .build();
 
       BruteForceIndexParams indexParams = new BruteForceIndexParams.Builder()
           .withNumWriterThreads(32)
           .build();
 
-      BruteForceIndex index = BruteForceIndex.newBuilder(resources)
-          .withDataset(dataset)
-          .withIndexParams(indexParams)
-          .build();
+      BruteForceIndex index;
+      boolean useNativeMemoryDataset = random.nextBoolean();
```

Comment:
I'm not sure where I stand on conditionally testing with `Dataset`.  Shouldn't we test both ways?

Reply:
The idea with randomized testing was to avoid loops to test for many variables, and run the test many times over and over to allow it to catch bugs. For this particular feature (Dataset vs float[][]), I think it makes sense to test for both. Fixing, thanks!